lib/src/main/java/com/scalar/admin/kubernetes/Pauser.java

```diff
@@ -77,67 +77,163 @@ public PausedDuration pause(int pauseDuration, @Nullable Long maxPauseWaitTime)
+    // Run unpause operation.
+    UnpauseFailedException unpauseFailedException = null;
+    try {
+      unpauseWithRetry(requestCoordinator, MAX_UNPAUSE_RETRY_COUNT);
+    } catch (Exception e) {
+      unpauseFailedException = new UnpauseFailedException("Unpause operation failed.", e);
+    }
 
+    // Get pods and deployment information after pause.
     TargetSnapshot targetAfterPause;
     try {
-      targetAfterPause = targetSelector.select();
+      targetAfterPause = getTarget();
     } catch (Exception e) {
       throw new PauserException(
           "Failed to find the target pods to examine if the targets pods were updated during"
               + " paused.",
           e);
     }
 
-    if (!target.getStatus().equals(targetAfterPause.getStatus())) {
-      throw new PauserException("The target pods were updated during paused. Please retry.");
+    // Prepare error messages for each process.
+    String unpauseErrorMessage =
+        String.format(
+            "Unpause operation failed. Scalar products might still be in a paused state. You"
+                + " must restart related pods by using the `kubectl rollout restart deployment"
+                + " %s` command to unpause all pods. ",
+            targetBeforePause.getDeployment().getMetadata().getName());
+    String pauseErrorMessage =
+        "Pause operation failed. You cannot use the backup that was taken during this pause"
+            + " duration. You need to retry the pause operation from the beginning to"
+            + " take a backup. ";
+    String statusCheckErrorMessage =
+        "Status check failed. You cannot use the backup that was taken during this pause"
+            + " duration. You need to retry the pause operation from the beginning to"
+            + " take a backup. ";
+    String statusDifferentErrorMessage =
+        "The target pods were updated during the pause duration. You cannot use the backup that"
+            + " was taken during this pause duration. ";
+    String unpauseFailedButBackupOkErrorMessage =
+        String.format(
+            "Note that the pause operations for taking backup succeeded. You can use a backup that"
+                + " was taken during this pause duration: Start Time = %s, End Time = %s. ",
+            startTime, endTime);
+
+    // Check if pods and deployment information are the same between before pause and after pause.
+    boolean compareTargetSuccessful;
+    StatusCheckFailedException statusCheckFailedException;
+    try {
+      compareTargetSuccessful = compareTargetStatus(targetBeforePause, targetAfterPause);
+    } catch (Exception e) {
+      statusCheckFailedException = new StatusCheckFailedException(statusCheckErrorMessage, e);
+      if (unpauseFailedException == null) {
+        throw statusCheckFailedException;
+      } else {
+        throw new UnpauseFailedException(unpauseErrorMessage, statusCheckFailedException);
+      }
     }
```

Comment:
It looks like `compareTargetStatus()` doesn't throw any exceptions. Do we still need this exception handling?

Maybe we need the exception handling for `targetAfterPause` here?
https://github.com/scalar-labs/scalar-admin-for-kubernetes/pull/33/files#diff-529e2e818d31d239079b031928623cab070acd5743e317b3d709c34e6e52b394R114-R121

Reply:
> It looks like `compareTargetStatus()` doesn't throw any exceptions. Do we still need this exception handling?

For safety, I think it would be better to handle exceptions here. There is a possibility that some unexpected exceptions might occur here, and it could cause the issue `keeping unpause state`. 

> Maybe we need the exception handling for `targetAfterPause` here? https://github.com/scalar-labs/scalar-admin-for-kubernetes/pull/33/files#diff-529e2e818d31d239079b031928623cab070acd5743e317b3d709c34e6e52b394R114-R121

As I mentioned in another comment, you're right. I will add the exception handling there. I overlooked it. Thank you for pointing it out!

However, as mentioned above, I think we can keep this exception handling here. What do you think?